plugins/woocommerce-admin/client/marketplace/contexts/product-list-context.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * External dependencies
+ */
+import { useState, useEffect, createContext } from '@wordpress/element';
+
+type ProductListItemType = {
+	title: string;
+	image: string;
+	excerpt: string;
+	link: string;
+	demo_url: string;
+	price: string;
+	hash: string;
+	slug: string;
+	id: number;
+	rating: number | null;
+	reviews_count: number | null;
+	vendor_name: string;
+	vendor_url: string;
+	icon: string;
+};
```

Comment:
I created this manually, looking at the `/wp-json/wccom-extensions/1.0/search` endpoint. Not sure if I got it 100%, especially around nullable values.

Reply:
I _think_ this is correct. Those two should be the only null values at this stage 🤔 – @simranvijwani may have a better idea here, since she worked on lots of API interaction initially.

Reply:
We've used `interface` instead of `type` elsewhere, I'm not sure if it matters. Might we use this for the product list items returned from the [featured](https://woocommerce.com/wp-json/wccom-extensions/2.0/featured?country=GB) endpoint? Those have a few extra properties, `label`, `averageRating` etc.